### PR TITLE
fix: detect "JavaScript heap out of memory" errors and fail job.

### PR DESCRIPTION
**Pull request by ebadiere**
Modified job to detect "JavaScript heap out of memory" errors.

**Related issue(s)**: #2262 

Fixes #

### CODE REVIEW

**Review by Nana-EC — DISMISSED** [review]

**Review by AlfredoG87 — DISMISSED** [review]
We might need to wait until we actually fix the issue to merge it, but otherwise LGTM

**Comment by konstantinabl** [comment]
Addressed here https://github.com/hashgraph/hedera-json-rpc-relay/pull/2305
So we can maybe close this PR : )

**Comment by ebadiere** [comment]
Replaced by https://github.com/hashgraph/hedera-json-rpc-relay/pull/2395

**Pull request closed without merging**
